src/axom/sina/config.hpp.in

```diff
@@ -0,0 +1,13 @@
+#ifndef SINA_CONFIG_HPP_
+#define SINA_CONFIG_HPP_
+
+/// \file
+
+#define SINA_FILE_FORMAT_VERSION_MAJOR @SINA_FILE_FORMAT_VERSION_MAJOR@
+#define SINA_FILE_FORMAT_VERSION_MINOR @SINA_FILE_FORMAT_VERSION_MINOR@
+#define SINA_FILE_FORMAT_VERSION_PATCH @SINA_FILE_FORMAT_VERSION_PATCH@
```

Comment:
It might be good to base this one in the role that the version number would play--we only _need_ it in the case that we made a formatting change that requires us to load a file differently (we're using it in a switch somewhere). So in terms of semantic versioning, a major version increase. Minor version increase would be purely informative ("this file might contain curve_sets", but adding curve sets doesn't break backwards compatibility). The version would be written into the file as one more piece of metadata, and loaded from it when we ingest, rather than something I imagine the codes themselves needing to check.

So formal in the sense of what the version numbers mean: I think semantic (possibly without patch number) would work fine.

Formal in the sense of when, where, and how this is changed: I can't think of a case at the moment where we'd change the version number without also changing the C++ lib, so I wouldn't foresee much issue in this one coming unsynced from the Python version except in the sense of who releases first.

Definitely happy to discuss this in more detail/at a meeting!